frame/quorum/src/lib.rs

```diff
@@ -687,27 +689,30 @@ pub mod pallet {
 
     // Record the vote in the storage
     fn commit_vote(who: T::AccountId, proposal_id: Hash, in_favour: bool) -> DispatchResult {
-      let block_number = T::Security::get_current_block_count();
+      let current_block = T::Security::get_current_block_count();
+      let maybe_proposal = Self::proposals()
+        .into_iter()
+        .find(|(id, _block_number, _proposal)| *id == proposal_id);
+      ensure!(maybe_proposal.is_some(), Error::<T>::ProposalDoesNotExist);
+
+      let (_id, proposal_block, _proposal) = maybe_proposal.unwrap();
       ensure!(
-        Self::proposals()
-          .into_iter()
-          .find(|(id, _, _)| *id == proposal_id)
-          .is_some(),
-        Error::<T>::ProposalDoesNotExist
+        current_block >= proposal_block,
+        Error::<T>::ProposalBlockIsInFuture
       );
```

Comment:
Well done again!

For better readability I would go with something like this;

```rust
let proposal_block = Self::proposals()
  .into_iter()
  .find(|(id, _block_number, _proposal)| *id == proposal_id)
  .ok_or(Error::<T>::ProposalDoesNotExist)?
  .1;

ensure!(
  current_block >= proposal_block,
  Error::<T>::ProposalBlockIsInFuture
);
```

Reply:
Thanks, it's updated to the PR now :)